src/RA_Dlg_RichPresence.cpp

```diff
@@ -20,7 +21,12 @@ INT_PTR CALLBACK Dlg_RichPresence::RichPresenceDialogProc(HWND hDlg, UINT nMsg,
 
         case WM_TIMER:
         {
-            std::wstring sRP = ra::Widen(g_RichPresenceInterpreter.GetRichPresenceString());
+            std::wstring sRP;
+            if (g_pCurrentGameData->GetGameID() != 0)
```

Comment:
What's the rationale behind extracting logic into the dialog window? That tends to be considered bad practice.

Reply:
It's not really a property of the interpreter. That should return whatever the currently loaded script generates. If no script is loaded, empty string is returned. 

Until we move to an MVVM or MVC architecture, I don't see any better place to put this. I will move it up one more step so its set once without using the timer.